scriptworker/worker.py

```diff
@@ -82,12 +82,13 @@
         context (scriptworker.context.Context): the scriptworker context.
     """
     tmp_gpg_home = get_tmp_base_gpg_home_dir(context)
-    lockfile = context.config['gpg_lockfile']
-    await run_loop(context)
-    await asyncio.sleep(context.config['poll_interval'])
-    if os.path.exists(tmp_gpg_home) and not os.path.exists(lockfile):
+    state = is_lockfile_present(context, "scriptworker", logging.DEBUG)
+    if os.path.exists(tmp_gpg_home) and state == "ready":
         overwrite_gpg_home(tmp_gpg_home, context.config['base_gpg_home_dir'])
         rm(tmp_gpg_home)
+        rm_lockfile(context)
```

Comment:
I wonder if the removal of the lockfile here should be protected under a `finally:` block. just in case we run into an exception while copying/rm gpg home.